app/src/androidTest/java/com/example/sharingang/MapActivityTest.kt

```diff
@@ -0,0 +1,67 @@
+package com.example.sharingang
+
+import android.os.Build
+import androidx.test.espresso.Espresso.onView
+import androidx.test.espresso.action.ViewActions.click
+import androidx.test.espresso.assertion.ViewAssertions.matches
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.espresso.matcher.ViewMatchers.withText
+import androidx.test.ext.junit.rules.ActivityScenarioRule
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import androidx.test.rule.GrantPermissionRule
+import androidx.test.uiautomator.UiDevice
+import androidx.test.uiautomator.UiSelector
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+
+@RunWith(AndroidJUnit4::class)
+class MapActivityTest {
+    @get:Rule
+    val activityRule = ActivityScenarioRule(MapActivity::class.java)
+    //@get:Rule 
+    //var permissionRule: GrantPermissionRule = GrantPermissionRule.grant(android.Manifest.permission.ACCESS_FINE_LOCATION)
+    // code taken from https://alexzh.com/ui-testing-of-android-runtime-permissions/
+    fun grantPermission() {
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+        if (Build.VERSION.SDK_INT >= 23) {
+            val allowPermission = UiDevice.getInstance(instrumentation).findObject(UiSelector().text(
+                    when {
+                        Build.VERSION.SDK_INT == 23 -> "Allow"
+                        Build.VERSION.SDK_INT <= 28 -> "ALLOW"
+                        Build.VERSION.SDK_INT == 29 -> "Allow only while using the app"
+                        else -> "While using the app"
```

Comment:
I would extract this out into a value

Reply:
I don't understand?

Reply:
e.g. `val searchText = when ...`

Reply:
Why?

Reply:
It's not all that nice to have this complicated expression inlined as a parameter to a function

Providing a value allows you to provide a name, and thus a bit of explanation as to what this string actually does.

Reply:
I tried but it doesn't work, I will leave it to you. Also, I don't think it is that complicated, I can add a comment if you want.